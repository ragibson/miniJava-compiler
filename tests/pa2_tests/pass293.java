// PA2 unary and binop precedence
class Unprecedented {
    void p(){
        int x = -1 + 2 * 3 / 4 > 5 >= 6 < 7 <= - -8 != 9 && !!false || true + new Unprecedented();
    }
}

