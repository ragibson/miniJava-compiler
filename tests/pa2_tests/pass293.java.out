********** miniJava Compiler (PA2) **********
Syntactic Analysis ... 
Syntactic Analysis OK
======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "Unprecedented" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   VOID BaseType
  .   .   "p" methodname
  .   .   ParameterDeclList [0]
  .   .   StmtList [1]
  .   .   . VarDeclStmt
  .   .   .   VarDecl
  .   .   .     INT BaseType
  .   .   .     "x" varname
  .   .   .   BinaryExpr
  .   .   .     "||" Operator
  .   .   .       BinaryExpr
  .   .   .         "&&" Operator
  .   .   .           BinaryExpr
  .   .   .             "!=" Operator
  .   .   .               BinaryExpr
  .   .   .                 "<=" Operator
  .   .   .                   BinaryExpr
  .   .   .                     "<" Operator
  .   .   .                       BinaryExpr
  .   .   .                         ">=" Operator
  .   .   .                           BinaryExpr
  .   .   .                             ">" Operator
  .   .   .                               BinaryExpr
  .   .   .                                 "+" Operator
  .   .   .                                   UnaryExpr
  .   .   .                                     "-" Operator
  .   .   .                                       LiteralExpr
  .   .   .                                         "1" IntLiteral
  .   .   .                                   BinaryExpr
  .   .   .                                     "/" Operator
  .   .   .                                       BinaryExpr
  .   .   .                                         "*" Operator
  .   .   .                                           LiteralExpr
  .   .   .                                             "2" IntLiteral
  .   .   .                                           LiteralExpr
  .   .   .                                             "3" IntLiteral
  .   .   .                                       LiteralExpr
  .   .   .                                         "4" IntLiteral
  .   .   .                               LiteralExpr
  .   .   .                                 "5" IntLiteral
  .   .   .                           LiteralExpr
  .   .   .                             "6" IntLiteral
  .   .   .                       LiteralExpr
  .   .   .                         "7" IntLiteral
  .   .   .                   UnaryExpr
  .   .   .                     "-" Operator
  .   .   .                       UnaryExpr
  .   .   .                         "-" Operator
  .   .   .                           LiteralExpr
  .   .   .                             "8" IntLiteral
  .   .   .               LiteralExpr
  .   .   .                 "9" IntLiteral
  .   .   .           UnaryExpr
  .   .   .             "!" Operator
  .   .   .               UnaryExpr
  .   .   .                 "!" Operator
  .   .   .                   LiteralExpr
  .   .   .                     "false" BooleanLiteral
  .   .   .       BinaryExpr
  .   .   .         "+" Operator
  .   .   .           LiteralExpr
  .   .   .             "true" BooleanLiteral
  .   .   .           NewObjectExpr
  .   .   .             ClassType
  .   .   .               "Unprecedented" Identifier
=============================================
